include/url.h

```diff
@@ -79,6 +79,11 @@ namespace Url
          */
         Url& abspath();
 
+        /**
+         * Evaluate 'other' relative to this URL, placing the result in this object.
+         */
+        Url& relative(const std::string& other);
```

Comment:
Yeah -- I think either we 1) make this mutating but change the sense of this (`base` would be the argument) or 2) make it return a _new_ URL. I think (1) makes the better choice